### ISSUE TITLE
RPC encoding regression in 2.6.1

### ISSUE THREAD

**Issue by hkalodner** [trufflesuite/ganache-core#467]
Version 2.6.1 upgraded the dependency on `web3-provider-engine` from version 14.1.0 to version 14.2.0. This version upgrade contains a regression (also reported in https://github.com/MetaMask/web3-provider-engine/issues/317).

The regression is a deviation from the rpc hex encoding specification described in https://github.com/ethereum/wiki/wiki/JSON-RPC#hex-value-encoding. `0` values are now encoded as `"0x"` when they should be `"0x0"`. This causes errors in rpc client libraries that expect strict correctness like go-ethereum.

 

## Expected Behavior
Calls to rpc methods which contain results including 0 values should correctly encode them as "0x0"

## Current Behavior
Calls to rpc methods which contain results including 0 values encode them as "0x"

## Possible Solution
This could be solved by either downgrading the web3-provider-engine dependency back to 14.1.0 or getting an upstream fix into web3-provider-engine and upgrading. If there are no other reasons why 14.2.0 is required, downgrading seems like the easiest immediate fix.

## Steps to Reproduce (for bugs)
The following code will print the difficulty returned by ganache-core as `0x` when running with 2.6.1 and `0x0` when running 2.6.0.
 ```javascript
const ganache = require("ganache-core");
const WebSocket = require('ws');

ganache.server({blockTime: 1}).listen(7545);
var client = new WebSocket("ws://localhost:7545/");
client.onopen = function(e){
    this.send('{"id":1337,"jsonrpc":"2.0","method":"eth_subscribe","params":["newHeads", {}]}');
}
client.onmessage = function(response,flags,number)  {
    try {
        console.log(JSON.parse(response.data).params.result.difficulty);
    } catch(e) {}
}
```

## Context
This came up trying to use ganache along with go-ethereum which forces strict adherence to the hex encoding in 
https://github.com/ethereum/go-ethereum/blob/dbb03fe9893dd19f6b1de1ee3b768317f22fd135/common/hexutil/json.go#L358.

Specifically calls to SubscribeNewHead result in a JSON unmarshalling error due to this problem.

## Your Environment

* Version used: 2.6.1
* Environment name and version (e.g. PHP 5.4 on nginx 1.9.1): Node v11.15.0
* Server type and version:
* Operating System and version: Mac 10.14.6
* Link to your project: https://github.com/OffchainLabs/arbitrum/blob/2e7e2299ef6f19167a8d8fbf042abe124fe13492/packages/arb-validator/ethbridge/ethbridge.go#L96

**Comment by kumavis** [comment]
patched in `web3-provider-engine@14.2.1`

**Comment by hkalodner** [comment]
Thanks @kumavis! It'd be great to get this dependency upgraded in this package. No wiggle room on the version pinning so it'll require an upgrade.
